app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -27,15 +30,33 @@ def registration_url(proxy = nil)
     "#{url}/register"
   end
 
+  def jwt_expiration_param
+    param = registration_params['jwt_expiration'] || 4
+    @jwt_expiration_param ||= begin
+      if param == 'unlimited'
+        0
+      else
+        Float(param).to_i
```

Comment:
We should catch an error when user sends invalid data directly to the API:

```bash
curl -X POST "http://localhost:3000/api/registration_commands" \
     -H "Content-Type: application/json" \
     --user admin:changeme \
     -d '{"registration_command": {"jwt_expiration": "unlimiteded"}}'
```
=>
```json
{
  "error": {"message":"invalid value for Float(): \"unlimiteded\""}
}
```

Reply:
Could you elaborate more? I tested this and as far as I know , we are already catching the error.  Am I missing something?

Reply:
validation is done in the following order :

1.  Float() is to invalidate Strings except 'unlimited'
2. `Invalid value %s for jwt_expiration. The value must be between 0 and 999999. 0 means 'unlimited'.` is basically to invalidate negative numbers

I think it makes sense to have 2 different errors for 2 different types of validations. I could however try to update it but that could mean changing the order(which is a lot of modification) or I could pass -1 as a value for strings, if it's absolutely required

Reply:
We probably talk about two different things, but my main concern is the message itself.
As a user, I don't want to get an `invalid value for Float()` error, I want to get easy to follow, descriptive error message, like `The value must be between 0 and 999999. 0 means 'unlimited'`.

Reply:
+1 

A single and easily understandable error message is always preferable when you consider the user-experience for our product.

`The value must be between 0 and 999999. Here 0 represents 'unlimited'.` --> This sounds great to me.